### ISSUE TITLE
win32 skip reporting is inconsistent in filepath tests

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5242]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)| @glyph reported|
|-|-|
|Trac ID|trac#5242|
|Type|enhancement|
|Created|2011-08-28 22:59:23Z|
|Branch|https://github.com/twisted/twisted/tree/win32-skip-5242|

Test output currently looks like this:

```
|| ===============================================================================
|| [SKIPPED]
|| Consider yourself lucky.
|| 
|| twisted.test.test_paths.FilePathTestCase.testInsecureWin32
|| twisted.test.test_paths.FilePathTestCase.testInsecureWin32Whacky
|| ===============================================================================
|| [SKIPPED]
|| True
|| 
|| twisted.test.test_paths.FilePathTestCase.test_getPermissions_Windows
|| twisted.test.test_paths.FilePathTestCase.test_statinfoBitsNotImplementedInWindows
|| -------------------------------------------------------------------------------
```

The message should probably be a bit clearer, but more importantly, the message should be the _same_ so that it's clear all four tests have been skipped for the same reason.

Attachments:

* [patch_5242.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/582/582e8cbec3ba2aab580eeec3c06842787758c34b/dd301a4cc2e31769e3f321a7d2c2ea19861654f1.patch) (1365 bytes) - added by moijes12 on 2012-02-01 17:55:35Z - Changes are made and results are tested.
* [patch_5242.2.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/582/582e8cbec3ba2aab580eeec3c06842787758c34b/e5dd09025e8d133f02cf75853754d12e7163bfe9.patch) (2575 bytes) - added by moijes12 on 2012-02-02 11:26:42Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__5242 5242
type__enhancement enhancement
reporter__glyph glyph
priority__low low
milestone__ 
branch__branches_win32_skip_5242 branches/win32-skip-5242
branch_author__moijes12 moijes12
status__closed closed
resolution__fixed fixed
component__core core
keywords__easy easy
time__1314572363000000 1314572363000000
changetime__1328451191000000 1328451191000000
version__None None
owner__thijs thijs
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="moijes12's avatar" src="https://avatars.githubusercontent.com/u/4566851?s=50" width="50" height="50">](https://github.com/moijes12)<a name="note_1"></a>|@moijes12 commented|
|-|-|

Output message changed to "Test will run only on Windows."

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_2"></a>|@thijstriemstra set owner to @moijes12|
|-|-|

Thanks. Other win32 related skip message I see on ubuntu 11.10 are:

```
===============================================================================
[SKIPPED]
This error lookup only works on Windows

twisted.test.test_strerror.ErrorFormatingTestCase.test_correctLookups
twisted.test.test_strerror.ErrorFormatingTestCase.test_fromEnvironment
===============================================================================
```

```
[SKIPPED]
_pollingfile is only avalable under Windows.

twisted.internet.test.test_pollingfile.TestPollableWritePipe.test_checkWorkUnicode
===============================================================================
```

```
[SKIPPED]
This error lookup only works on Windows

twisted.test.test_strerror.ErrorFormatingTestCase.test_correctLookups
twisted.test.test_strerror.ErrorFormatingTestCase.test_fromEnvironment
```

Can you also unify those with the same skip message?

**Comment by twisted-trac** [comment]
|[<img alt="moijes12's avatar" src="https://avatars.githubusercontent.com/u/4566851?s=50" width="50" height="50">](https://github.com/moijes12)<a name="note_3"></a>|@moijes12 removed owner|
|-|-|

The new patch contains the previous change as well as the requested one.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_4"></a>|@thijstriemstra commented|
|-|-|

(In [33509]) Branching to 'win32-skip-5242'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_5"></a>|@thijstriemstra commented|
|-|-|

(In [33510]) Apply patch_5242.2.patch, with some coding standard fixes. refs [#5242](https://github.com/twisted/twisted/issues/5242)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_6"></a>|@thijstriemstra set owner to @thijstriemstra<br>@thijstriemstra set status to `assigned`|
|-|-|

Applied the patch, result looks good:

```
[SKIPPED]
Test will run only on Windows.

twisted.internet.test.test_pollingfile.TestPollableWritePipe.test_checkWorkUnicode
twisted.test.test_paths.FilePathTestCase.testInsecureWin32
twisted.test.test_paths.FilePathTestCase.testInsecureWin32Whacky
twisted.test.test_paths.FilePathTestCase.test_getPermissions_Windows
twisted.test.test_paths.FilePathTestCase.test_statinfoBitsNotImplementedInWindows
twisted.test.test_strerror.ErrorFormatingTestCase.test_correctLookups
twisted.test.test_strerror.ErrorFormatingTestCase.test_fromEnvironment
===============================================================================
```

I made some coding standard updates and added a news file, going to merge it.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_7"></a>|@thijstriemstra set status to `closed`|
|-|-|

(In [33511]) Merge win32-skip-5242: Updated the skip message for win32 filepath tests so it's clear they were skipped for the same reason.

Author: moijes12
Reviewer: thijs
Fixes: [#5242](https://github.com/twisted/twisted/issues/5242)
